src/vuex/modules/index/types.ts

```diff
@@ -1,11 +1,142 @@
+import _ from 'lodash'
+
+export class Index {
+  private _name: string
+  loading = false
+  collections?: Collection[]
+
+  constructor(name: string) {
+    this._name = name
+  }
+
+  get name() {
+    return this._name
+  }
+
+  get collectionsCount(): number | undefined {
+    return this.collections?.length
+  }
+
+  public initCollections(collections: Collection[]) {
+    this.collections = collections
+  }
+
+  public addCollection(collection: Collection) {
+    if (!this.collections) {
+      throw new Error(
+        'Unable to perform operations, the collection list is not yet initialized'
+      )
+    }
+
+    this.collections.push(collection)
+  }
+
+  public removeCollection(collection: Collection) {
+    if (!this.collections) {
+      throw new Error(
+        'Unable to perform operations, the collection list is not yet initialized'
+      )
+    }
+
+    this.collections = this.collections.filter(
+      el => el.name !== collection.name
+    )
+  }
+
+  public getOneCollection(collectionName: string) {
+    if (!this.collections) {
+      throw new Error(
+        'Unable to perform operations, the collection list is not yet initialized'
+      )
+    }
+
+    return this.collections.find(el => el.name === collectionName)
+  }
+
+  public doesCollectionExist(collectionName: string): boolean {
+    if (!this.collections) {
+      throw new Error(
+        'Unable to perform operations, the collection list is not yet initialized'
+      )
+    }
+
+    return this.collections.find(el => el.name === collectionName)
+      ? true
+      : false
+  }
+}
+
+export class Collection {
+  private _name: string
+  private _type: CollectionType
+  mapping?: object
+  dynamic?: string
+
+  constructor(name: string, type: CollectionType) {
+    this._type = this.findType(name, type)
+    this._name = name
+  }
+
+  get name(): string {
+    return this._name
+  }
+
+  get type(): CollectionType {
+    return this._type
+  }
+
+  public isRealtime(): boolean {
+    return this._type === CollectionType.REALTIME
+  }
+
+  private findType(name: string, type: string) {
```

Comment:
I would have created a `enum` type, rather than making all these checks, but if you wanna keep it like this...
```suggestion
  private checkType(name: string, type: string) {
```

Reply:
oh... I see the `enum` right below... Doesn't the language check the enum for you?